ui/src/shared/components/inlineLabels/InlineLabelsEditor.tsx

```diff
@@ -160,35 +182,63 @@ class InlineLabelsEditor extends Component<Props, State> {
   private handleInputChange = async (
     e: ChangeEvent<HTMLInputElement>
   ): Promise<void> => {
-    const {availableLabels} = this
-    let selectedItemID = this.state.selectedItemID
     const searchTerm = e.target.value
+    const filteredLabels = this.filterLabels(searchTerm)
 
-    const selectedItemIDAvailable = availableLabels.find(
-      al => al.name === selectedItemID
-    )
+    if (filteredLabels.length) {
+      const selectedItemID = filteredLabels[0].id
```

Comment:
It always feels weird when I see code that just selects the first on from the list, there is no clear definition as to why the first element gets special status over the whole list.